File: src/randomizer/card-supply-ban.ts
import {Cards} from "../utils/cards";
import type {SupplyBan} from "./supply-ban";
import type {SupplyCard} from "../dominion/supply-card";

export class CardSupplyBan implements SupplyBan {
  constructor(readonly cardIds: string[]) {
  }

  getBannedCards(cards: SupplyCard[]): SupplyCard[] {
    return cards.filter(Cards.filterByIncludedIds(this.cardIds));
  }
}

File: flutter/dominion_randomizer/lib/services/randomizer/card_supply_ban.dart
import 'package:dominion_randomizer/models/card.dart';

import 'supply_ban.dart';

class CardSupplyBan implements SupplyBan {
  final Set<String> cardIds;

  CardSupplyBan(this.cardIds);

  @override
  Iterable<Card> getBannedCards(Iterable<Card> cards) {
    return cards.where((card) => cardIds.contains(card.id));
  }
}
